web/src/app/admin/admin-message-logs/util.ts

```diff
@@ -0,0 +1,23 @@
+import { DateTime, Duration } from 'luxon'
+import { useURLParams } from '../../actions'
+
+export type MessageLogsParams = {
+  search: string
+  start: string
+  end: string
+  graphInterval: string
+}
+
+export function useMessageLogsParams(): [
+  MessageLogsParams,
+  (params: Partial<MessageLogsParams>) => void,
+] {
+  const end = DateTime.now().startOf('hour').plus({ hour: 1 })
```

Comment:
This function in its own file kind of seems like overkill since they are both just grabbing some items from the URL

Reply:
For this to work correctly, all three components must have completely identical defaults, which is why I made the shared function. It would also be hard to debug if it were to get out of sync.

A shared `util.ts` is the convention we've used elsewhere for shared functions/hooks.

---

Elsewhere we've read the parameters in a single place and then passed the values down to other components as props -- that could be an alternative.